### DISCUSSION TITLE
Drift modular build runner issue with @UseRowClass?

### DISCUSSION THREAD

**Discussion by bZichett** [simolus3/drift#2260] [Q&A] [answered]
After having a stable non-modular configuration, I am doing a quick experiment with modular build runner and running into an issue after clearing all issues in my source code.  

The Table:
```
@UseRowClass(Tag) // Note: Uses built_value, another additional complexity to this modular experiment
class Tags extends Table {
   ...
```

_Note 1: I can share the build runner build.yml configuration but its directly utilizing the one in the modular docs - this might be the issue (different code paths - `@UseRowClass` annotation?)_

The file writer's output for this example table is:
`class $TagsTable extends Tags with i0.TableInfo<$TagsTable, i1.Tag> {`

This is new.  It's usually Table (before modular.)  It can be manually fixed by replacing with:
`class $TagsTable extends i0.Table with i0.TableInfo<$TagsTable, i1.Tag> {`

_Note 2: Tags => i0.Table_

The [example](https://github.com/simolus3/drift/blob/develop/examples/modular/lib/src/posts.drift) in the repository for modular code uses the sql.drift files for defining tables and I was wondering if the above use case was supported / tested or there is actually a configuration issue on my end.  I looked at the table writer code in the drift_dev repo and was trying to figure out why this particular variable is utilized differently.  The particular line is here: 
https://github.com/simolus3/drift/blob/develop/drift_dev/lib/src/writer/tables/table_writer.dart#L350

And it's computed using to `tableDslName`:

https://github.com/simolus3/drift/blob/develop/drift_dev/lib/src/writer/tables/table_writer.dart#L345

Changing this line in the library to always use `emitter.drift('Table')` - and _never_ `table.definingDartClass`  fixes the issue on a new build, but I am likely missing some intent in the design.  Although the code now compiles and all is well.  

In advance; I much appreciate the direction /help and I must say this has been a spectacular library so far @simolus3 :clap: :+1:

**Comment by simolus3** [comment, marked as answer]
Thanks for trying out modular code generation and for the feedback!

The intention here is that the generated table classes should extend the defined DSL classes. This is also be the case without non-modular code generation, like here:

https://github.com/simolus3/drift/blob/6b4a186b15da8392e853359e6c6d642aa8080b82/drift/example/main.g.dart#L6-L7

The problem is that we just treat the name of the original class as a string, assuming that it will always be in the same library as the generated code. This assumption breaks when using modular code generation, which generates independent Dart files without a `part of` statement. So the fix is as simple as importing the original class, so that we generate something like

```dart
import 'path_to_your_tags_table.dart' as i2;

class $TagsTable extends i2.Tags with i0.TableInfo<$TagsTable, i1.Tag> {
```

And that is exactly what I've done in 6b4a186b15da8392e853359e6c6d642aa8080b82, which should fix the issue.

**Reply by bZichett** [reply]
Thanks for the incredibly speedy fix and explanation.  

> Thanks for trying out modular code generation and for the feedback!

My pleasure.  Keep up the great work!
